src/libra/engine/forward_runner.py

```diff
@@ -0,0 +1,121 @@
+import ast
+from queue import Queue
+from pip._vendor.colorama import Style
+from typing import Set, Dict 
+import time
+
+from apronpy.manager import PyManager
+from apronpy.box import PyBoxMPQManager
+from apronpy.texpr1 import PyTexpr1
+from apronpy.environment import PyEnvironment
+from apronpy.var import PyVar
+ 
+from libra.core.cfg import Node, Function, Activation
+from libra.engine.bias_analysis import AbstractDomain
+from libra.core.expressions import VariableIdentifier
+from libra.engine.runner import Runner
+from libra.abstract_domains.deeppoly_domain import DeepPolyState
+from libra.abstract_domains.neurify_domain import NeurifyState
+from libra.abstract_domains.interval_domain import BoxState
+from libra.abstract_domains.symbolic1_domain import Symbolic1State
+from libra.abstract_domains.symbolic2_domain import Symbolic2State
+from libra.core.statements import Assignment, Lyra2APRON
+from libra.frontend.cfg_generator import ast_to_cfg
+from libra.engine.forward import ForwardInterpreter, ActivationPatternForwardSemantics
+
+class ForwardRunner(Runner):
```

Comment:
This does not quite belong to Libra but we can keep it. For consistency with the other runner, I would rather rename it `ForwardAnalysis`, though.